### ISSUE TITLE
Check empty ID attribute of LoadFont and Font

### ISSUE THREAD

**Issue by matmat** [Ymagis/ClairMeta#134]
I have a subtitle XML with the following (pretty-printed)

```
<?xml version="1.0" encoding="UTF-8"?>
<dcst:SubtitleReel xmlns:dcst="http://www.smpte-ra.org/schemas/428-7/2010/DCST" xmlns:xs="http://www.w3.org/2001/XMLSchema">
(...)
  <dcst:LoadFont ID="">urn:uuid:47471663-927b-4008-957a-faf6fd14be32</dcst:LoadFont>
  <dcst:SubtitleList>
    <dcst:Font AspectAdjust="1.0" Color="FFFFFFFF" Effect="border" EffectColor="FF000000" ID="" Italic="no" Script="normal" Size="42" Underline="no" Weight="normal">
(...)
```

This gives the following error on a Barco ICMP player:
```
Dec  9 12:53:12 icmp-0e26fd user.err GIP: XML Parse: ERROR on line 2 , column 586
Dec  9 12:53:12 icmp-0e26fd user.err GIP: XML Parse: Error string = parsing aborted
Dec  9 12:53:12 icmp-0e26fd user.err GIP: XML Parse: Error Info string = Bad LoadFont Tag
Dec  9 12:53:12 icmp-0e26fd user.err GIP: XML Parse: Parser returned with error.
Dec  9 12:53:12 icmp-0e26fd user.err GIP: The error occured in: /tmp/subtitles/published/980d2baf-4c93-4c44-ab78-bfcd8fd3c6b7/980d2baf-4c93-4c44-ab78-bfcd8fd3c6b7.xml
Dec  9 12:53:12 icmp-0e26fd user.err GIP: SubtitleError--Subtitle-XML Parsing Error
Dec  9 12:53:12 icmp-0e26fd user.info GIP: Flush Overlay
Dec  9 12:53:12 icmp-0e26fd user.err SMS: Player- Player subtitles: activate gip subtitles failed: 0
Dec  9 12:53:12 icmp-0e26fd user.err SMS: Player- Player error: player - subtitles validation error - cannot play
Dec  9 12:53:12 icmp-0e26fd user.info SMS: Report- |SHOW-SELECT| [BomberHarris_SHR-1-30_F-143_EN-SV_20_2K_20191208_SMPTE_OV]
Dec  9 12:53:12 icmp-0e26fd user.debug SMS: Player- starting streams at frame id: 0 - streams: AV
Dec  9 12:53:12 icmp-0e26fd user.debug SM: IMB Controller- Sync Payload Updated (frameId: 0, status: 0, delay: 0, offset: 0)
Dec  9 12:53:12 icmp-0e26fd user.debug SMS: Player- Player subtitles: updating gip timeline: 00:00:00:000
Dec  9 12:53:13 icmp-0e26fd user.info GIP: Overlay_Flush(): free buffers = 3
Dec  9 12:53:13 icmp-0e26fd user.info GIP: Overlay_Flush() done
```

Also attaching the subtitle XML in full:

[980d2baf-4c93-4c44-ab78-bfcd8fd3c6b7.zip](https://github.com/Ymagis/ClairMeta/files/3965132/980d2baf-4c93-4c44-ab78-bfcd8fd3c6b7.zip)

It would be nice if clairmeta would check empty ID attributes for LoadFont and Font. Should probably also be done for DCSubtitle v1.0 and v1.1 as well as 428-7:2014.

**Comment by remia** [comment]
Hi @matmat,

Thanks for reporting, indeed it appears that the SMPTE XSD implementing 428-7:2014 is not properly checking this. I'm not sure if we use the latest version of that schema, in the mean time I added an explicit check in here #136.

For Interop, the schema (that you wrote I think!) is already checking this and correctly raise an error in this case according to my test. As for the empty Font ID attribute, I think  it'll not be necessary to add a check for this because we will catch empty LoadFont ID and there is a check for matching LoadFont ID and Font ID reference https://github.com/Ymagis/ClairMeta/blob/develop/clairmeta/dcp_check_subtitle.py#L295.

The point I'm unclear is if that could break compatibility with older 428-7 versions (not sure we support that kind of SMPTE standard versioning in the first place).

**Comment by remia** [comment]
Fixed in #136 and #139
